release/Makefile

```diff
@@ -0,0 +1,54 @@
+# Needed for glob expansion.
+SHELL = /bin/bash
+.SHELLFLAGS = -O extglob -c
+
+GIT_TAG = $(shell git describe --dirty --always --tags)
+ifneq ($(DRY_RUN),)
+SNAPSHOT_FLAGS = --snapshot --skip-publish --rm-dist
+TAG ?= $(GIT_TAG)
+endif
+
+.PHONY: all
+all: release
+
+.PHONY: check_tag
+check_tag:
+ifeq ($(TAG),)
+	$(error "TAG must be set to a release tag")
+endif
+
+.PHONY: changelog
+changelog: check_tag ## Generate the changelog.
+	mkdir -p changelog/generated
```

Comment:
@joelanford `goreleaser` needs a file containing release notes if we're going to use a custom changelog generator, so the idea is to create a new changelog file per release under `changelog/generated`, much like we do with migration guides. This file will be committed in the release commit prior to running `goreleaser`, much as we do with `CHANGELOG.md` now.

Reply:
> This file will be committed in the release commit prior to running goreleaser, much as we do with CHANGELOG.md now.

Sounds good while we have the need for a pre-release commit.

Thinking more in the future, if we're able to get rid of the pre-release commit would it be possible to generate it on the fly during the tag CI run just before running goreleaser? It might take some non-trivial changes to our current changelog fragment and generation process, but I think that would be on the table if we could get to the point of not needing pre-release commits.

Reply:
>if we're able to get rid of the pre-release commit would it be possible to generate it on the fly during the tag CI run just before running goreleaser

We could get rid of committing the changelog entirely, if that's desirable, right now (I think it is since we publish release notes).

Reply:
I take ^ back since we still need to remove fragments in the pre-release commit.

Reply:
Yeah, maybe let's make a separate issue to investigate options for changelog/release notes generation without a pre-release commit.